climpred/utils.py

```diff
@@ -84,13 +85,20 @@ def reduce_time_series(forecast, reference, nlags):
         reference (`xarray` object): reference being compared to (for skill,
                                      persistence, etc.)
         nlags (int): number of lags being computed
-
     Returns:
        forecast (`xarray` object): prediction ensemble reduced to
        reference (`xarray` object):
     """
+
     imin = max(forecast.time.min(), reference.time.min())
-    imax = min(forecast.time.max(), reference.time.max() - nlags)
+    offset_args_dict = get_lead_pdoffset_args(getattr(forecast['lead'], 'units'), nlags)
+
+    ref_dates = pd.to_datetime(
+        reference.time.dt.strftime('%Y%m%d 00:00')
```

Comment:
Since you use previously, maybe you can define it in constants? `'%Y%m%d 00:00'`